bot/extensions/clashofcode/commands.py

```diff
@@ -0,0 +1,290 @@
+import asyncio
+import re
+import time
+
+import aiohttp
+import discord
+from discord import Forbidden, HTTPException, NotFound, app_commands
+from discord.ext import commands
+
+from bot import core
+from bot.config import settings
+from bot.extensions.clashofcode.utils import coc_helper
+
+REGEX = re.compile(r"https://www.codingame.com/clashofcode/clash/([0-9a-f]{39})")
+API_URL = "https://www.codingame.com/services/ClashOfCode/findClashByHandle"
+
+
+def em(mode, players):
+    embed = discord.Embed(title="**Clash started**")
+    embed.add_field(name="Mode", value=mode, inline=False)
+    embed.add_field(name="Players", value=players)
+    return embed
+
+
+class ClashOfCode(commands.GroupCog, group_name="coc"):
+    session_commands = app_commands.Group(
+        name="session",
+        description="Commands for clash of code sessions",
+    )
+
+    def __init__(self, bot: core.DiscordBot):
+        self.bot = bot
+        self.session = False
```

Comment:
Is this attribute ever used?

Reply:
its being set to indicate if an active session is ongoing but when checking for an ongoing session it uses session_message instead to see if it is not 0. So this variable could be removed or the session checks changes to use self.session instead. @SylteA thoughts?